cylc/flow/xtrigger_mgr.py

```diff
@@ -371,12 +371,10 @@ def check_xtriggers(
             db_update_func: method to update xtriggers in the DB
 
         """
-        satisfied = set(
-            [
-                itask for itask in itasks
-                if itask.state.xtriggers_all_satisfied()
-            ]
-        )
+        satisfied = {
+            itask for itask in itasks
+            if itask.state.xtriggers_all_satisfied()
+        }
```

Comment:
Saves building and destroying intermediate list.

Reply:
:facepalm: